docarray/array/mixins/evaluation.py

```diff
@@ -14,31 +14,64 @@ class EvaluationMixin:
 
     def evaluate(
         self,
-        other: 'DocumentArray',
         metric: Union[str, Callable[..., float]],
+        ground_truth: Optional['DocumentArray'] = None,
```

Comment:
please avoid breaking the interface, `ground_truth` can be added as an alias, u can have a `deprecation` decorator that has `other` equivalent to `groundtruth`

Reply:
The problem is, that the function has atm two mandatory attributes `other` and `metric` (in this order). It does not make sense to keep `other` mandatory if the DocumentArray and its matches itself have labels. So I have to make it optional. If I do this I can not preserve the order since `metric` as a mandatory attribute has to be placed before `other`.

Reply:
The only way how it might work is to make metric optional as well by setting a default metric.

Reply:
other and ground_truth need to be equivalent, this 100% sure

Reply:
this can be achieved by a decorator

Reply:
@bwanglzu , the function in docarray.math is called `precision_at_k`, setting a `k` is possible but not mandatory since the matches usually contain only the Top-K

Reply:
Just for documentation: we agreed on checking the type of the first argument. If it is a `DocumentArray` we set ground_truth to metric and raise a deprecation warning. We also handle cases where someone provides a other keyword argument and raise a deprecation warning.

Reply:
oh i remembered